File: presentation/src/main/java/com/github/anrimian/musicplayer/ui/library/compositions/LibraryCompositionsView.java
package com.github.anrimian.musicplayer.ui.library.compositions;

import com.github.anrimian.musicplayer.domain.models.order.Order;
import com.github.anrimian.musicplayer.ui.library.common.compositions.BaseLibraryCompositionsView;

import moxy.viewstate.strategy.alias.OneExecution;

public interface LibraryCompositionsView extends BaseLibraryCompositionsView {

    @OneExecution
    void showSelectOrderScreen(Order order);

}

File: app/src/main/java/com/github/anrimian/musicplayer/ui/library/compositions/LibraryCompositionsView.kt
package com.github.anrimian.musicplayer.ui.library.compositions

import com.github.anrimian.musicplayer.domain.models.composition.Composition
import com.github.anrimian.musicplayer.domain.models.order.Order
import com.github.anrimian.musicplayer.ui.library.common.compositions.BaseLibraryCompositionsView
import moxy.viewstate.strategy.alias.OneExecution

interface LibraryCompositionsView : BaseLibraryCompositionsView<Composition> {

    @OneExecution
    fun showSelectOrderScreen(order: Order)

}